inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImplIntegrationTest.java

```diff
@@ -316,7 +317,12 @@ public void nonempty_WithEmptyKnowledgeBase_ShouldReturnTrue() {
         sut.registerKnowledgeBase(kb, sut.getNativeConfig());
         sut.defineBaseProperties(kb,false);
         List<KBHandle> listProperties = sut.listProperties(kb, true);
+        Stream<String> listIdentifier = listProperties.stream().map(KBHandle::getIdentifier);
```

Comment:
Please split your tests into three parts:
1) Setup
2) Action you want to test
3) Assertions

Reply:
Because we add the properties in accordance to KB schema settings, when we update KB schema (supposedly form OWL to wikidata) , we need to delete the properties from OWL and and then add default properties from wikidata otherwise it will add up to list of  default properties

Reply:
We needed kbUpdate flag to tell the method explicitly that, the call has been made from the update and we need to delete the old property values now.

Reply:
The name of the fields at times are the same, so we cannot just update the properties. It can cause duplicate name on property list and we definitely dont want that.

Reply:
Yes, but you delete all properties from what I see.

Reply:
> Please split your tests into three parts:
> Setup
> Action you want to test
> Assertions

What do u mean to say here?

Reply:
Look at the other tests and how they are structured. They are written in three blocks.

Reply:
> Yes, but you delete all properties from what I see. That means, if you call this method, ALL properties are gone.

Yeah, when u change ur schema, it intializes the KB again along with the default properties and reads the local/remote ontology again

Reply:
Ahhan , you mean spaces for blocks . Okay

Reply:
`listProperties` returns ALL properties, not only ontology properties.

Reply:
`if (kb.getType() == RepositoryType.LOCAL) {
           kbService.defineBaseProperties(kb,true);
`
and we call the method only for LOCAL type KB. 

So, what happens now is we didn't had the subClass as property anymore and even when we have an empty KB we dont see anything on KB page. To sort this out, we had to add this idea that when we initialize KB we should add the default properties which were earlier introduced by inferencer

Reply:
> You would also need to change for readOnly in the first block.

I didn't get you. Can you give the reference

Reply:
https://github.com/inception-project/inception/pull/449/files#diff-2c7aa5a57022089e3825589d1f7534c1R165